service/openapi.yaml

```diff
@@ -0,0 +1,285 @@
+          ETag:
+            $ref: '#/components/schemas/flagChange/properties/ETag'
+    bulkEvaluationFailure:
+      description: Bulk evaluation failure response
+      properties:
+        errorCode:
+          type: string
+          description: Error code specific to the bulk evaluation.
+        errorDetails:
+          type: string
+          description: Optional error details description for logging or other needs
+      required:
+        - errorCode
+    flagChangeFailure:
+      description: Bulk evaluation failure response
+      properties:
+        errorCode:
+          type: string
+          description: Error code specific to the bulk evaluation.
+        errorDetails:
+          type: string
+          description: Optional error details description for logging or other needs
+      required:
+        - errorCode
+    evaluationRequest:
+      description: Flag evaluation request
+      properties:
+        flagType:
+          $ref: '#/components/schemas/flagType'
+        context:
+          $ref: '#/components/schemas/context'
+    success:
+      description: Flag evaluation success response.
+      properties:
+        key:
+          $ref: '#/components/schemas/key'
+        value:
+          type: object
+          description: Flag evaluation result
+        reason:
+          type: string
+          description: Reason matching OpenFeature reason
+        variant:
+          type: string
+          description: Variant of the evaluated flag value
+        metadata:
+          type: object
+          description: Arbitrary metadata supporting flag evaluation
+      required:
+        - key
+        - value
+    failure:
+      description: Flag evaluation failure response
+      properties:
+        key:
+          $ref: '#/components/schemas/key'
+        errorCode:
+          type: string
+          enum: [FLAG_NOT_FOUND,PARSE_ERROR,TYPE_MISMATCH,TARGETING_KEY_MISSING,INVALID_CONTEXT,GENERAL]
+          description: OpenFeature compatible error code. See https://openfeature.dev/specification/types#error-code
+        errorDetails:
+          type: string
+          description: Optional error details description for logging or other needs
+      required:
+        - key
+        - errorCode
+    key:
+      type: string
+      description: Feature flag key
+      example: my-flag
+    flagType:
+      type: string
+      enum:
+        - string
+        - boolean
+        - integer
+        - float
+        - object
+      description: Type to evaluate the given flag
+    flags:
+      type: array
+      description: (optional) List of flags to be evaluated, if not set all the flag available will be evaluated against the context.
+      items:
+        $ref: '#/components/schemas/key'
+      example:
+        - my-flag
+        - my-other-flag
+    context:
+      type: object
+      description: Context information for flag evaluation
```

Comment:
Is this optional? My OpenAPI is a bit rusty... I would encourage that we make it optional, I think, consistent with the OpenFeature API.

Reply:
Here we only define reusable schemas. In OpenAPI unless stated all parameters are optional. We have to use `required` keys if we enforce mandatory fields. See this for example [1]

[1] - https://github.com/open-feature/protocol/blob/0e4d0ddf4cc3b953695d8bd63670f95420b3d139/service/openapi.yaml#L228-L230